docs/packaging.md

```diff
@@ -0,0 +1,120 @@
+# Packaging
```

Comment:
I don't think this name is very descriptive. What is the goal of a person opening this chapter? 
Packaging looks much similar to "extending".